systemtest/src/test/java/io/strimzi/systemtest/security/SecurityST.java

```diff
@@ -675,76 +676,115 @@ void testAutoRenewCaCertsTriggerByExpiredCertificate(ExtensionContext extensionC
-            .withClusterName(clusterName)
+            .withTopicName(testStorage.getTopicName())
+            .withNamespaceName(testStorage.getNamespaceName())
+            .withClusterName(testStorage.getClusterName())
             .withMessageCount(MESSAGE_COUNT)
-            .withKafkaUsername(user.getMetadata().getName())
+            .withKafkaUsername(testStorage.getUserName())
             .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
             .build();
 
-        Map<String, String> kafkaPods = PodUtils.podSnapshot(namespaceName, kafkaSelector);
+        Map<String, String> kafkaPods = PodUtils.podSnapshot(testStorage.getNamespaceName(), kafkaSelector);
 
-        LOGGER.info("Annotate secret {} with secret force-renew annotation", secretName);
-        Secret secret = new SecretBuilder(kubeClient(namespaceName).getSecret(namespaceName, secretName))
-            .editMetadata()
-                .addToAnnotations(Ca.ANNO_STRIMZI_IO_FORCE_RENEW, "true")
-            .endMetadata().build();
-        kubeClient(namespaceName).patchSecret(namespaceName, secretName, secret);
+        CertificateAuthority newCAValidity = new CertificateAuthority();
+        newCAValidity.setRenewalDays(150);
+        newCAValidity.setValidityDays(200);
 
-        LOGGER.info("Wait until maintenance windows starts");
-        LocalDateTime finalMaintenanceWindowStart = maintenanceWindowStart;
-        TestUtils.waitFor("maintenance window start",
-            Constants.GLOBAL_POLL_INTERVAL, Duration.ofMinutes(maintenanceWindowDuration).toMillis() - 10000,
-            () -> LocalDateTime.now().isAfter(finalMaintenanceWindowStart));
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(
+            testStorage.getClusterName(),
+            k -> {
+                k.getSpec().setClusterCa(newCAValidity);
+                k.getSpec().setClientsCa(newCAValidity);
+            },
+            testStorage.getNamespaceName()
+        );
 
-        LOGGER.info("Maintenance window starts");
+        TestUtils.waitFor("Wait for reconciliation to happen and make sure neither Kafka nor CA certs were renewed (as we're outside of maintenance time window).",
+            Constants.GLOBAL_POLL_INTERVAL_MEDIUM, Constants.SAFETY_RECONCILIATION_INTERVAL,
+            () -> {
+                Secret secretCaCluster = kubeClient(testStorage.getNamespaceName()).getSecret(testStorage.getNamespaceName(), clusterSecretName);
+                Secret secretCaClients = kubeClient(testStorage.getNamespaceName()).getSecret(testStorage.getNamespaceName(), clientsSecretName);
+                return secretCaCluster.getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CA_CERT_GENERATION).equals("0")
+                    && secretCaClients.getMetadata().getAnnotations().get(Ca.ANNO_STRIMZI_IO_CA_CERT_GENERATION).equals("0")
+                    && kafkaPods.equals(PodUtils.podSnapshot(testStorage.getNamespaceName(), kafkaSelector));
+            }
+        );
```

Comment:
You just changed the CR configuration. That should bump the generation. So you can simply wait for the observed generation to bump as well to indicate the reconciliation is complete. nd then you can simply get and assert the secret. No need to poll the secret for not changing.